evalml/automl/automl_search.py

```diff
@@ -180,11 +182,14 @@ def search(
         "patience": patience,
         "tolerance": tolerance,
         "verbose": verbose,
+        "problem_configuration": problem_configuration,
+        "n_splits": n_splits,
```

Comment:
Rather than threading this parameter to down to `AutoMLSearch` we should just call `make_data_splitter` and pass in as the `data_splitter` parameter?